frontend/src/components/ColumnChartPreview.tsx

```diff
@@ -14,12 +14,19 @@ type Props = {
   title: string;
   summary: string;
   columns: Array<string>;
-  data?: Array<object>;
+  data?: Array<any>;
   summaryBelow: boolean;
 };
 
 const ColumnChartPreview = (props: Props) => {
   const colors = useColors(props.columns.length);
+  const xAxisType = useCallback(() => {
+    return props.data &&
+      props.data.every((row) => typeof row[props.columns[0]] === "number")
+      ? "number"
+      : "category";
+  }, [props]);
```

Comment:
It's more efficient if we set specifically `data` and `columns` as dependencies of the `useCallback` instead of the entire `props` object. We can avoid some unnecessary re-renders by doing something like this: 

```
const { data, columns } = props;
const xAxisType = useCallback(() => {
    return data &&
      data.every((row) => typeof row[columns[0]] === "number")
      ? "number"
      : "category";
  }, [data, columns]);
```

Reply:
Sounds good, done